### DISCUSSION TITLE
API rule - none contain

### DISCUSSION THREAD

**Discussion by f11xter** [pocketbase/pocketbase#3247] [Q&A] [answered]
I want to allow deleting a record only when its relation has no entries in another table.

I tried

```
!(@collection.other.id ?= id)
```

and

```
(@collection.other.id ?= id) = false
```

Is something like this supported?

**Comment by ganigeorgiev** [comment, marked as answer]
Using an API rule like `@collection.other.someRelField != id` should work for non-admin requests and it roughly translates to:
_Require all records from collection "other" to have "someRelField" different than the current id_.

**But if `someRelField` in your case is a single `relation` field you may not need the above API rule.**
Instead you can mark the relation field **Nonempty** (_and set "Cascade delete:false", if not already_) and it should throw an error when attempting to delete the record that references it (_since this is field constraint it will work also when attempting to delete from the Admin UI_).

**Reply by f11xter** [reply]
Amazing thank you!
